src/dask_awkward/lib/optimize.py

```diff
@@ -292,7 +281,7 @@ def rewrite_layer_chains(dsk: HighLevelGraph, keys: Any) -> HighLevelGraph:
     chains = []
     deps = copy.copy(dsk.dependencies)
 
-    required_layers = {k[0] for k in keys}
+    required_layers = {k[0] for k in keys if isinstance(k, tuple)}
```

Comment:
@douglasdavis this is a big change. I am not sure, given Dask's type hints, whether we can safely assume that a key is always a tuple, can we? But it seems like this code does. I have too little domain specific knowledge here to know, so I encoded the assertion in an isinstance (rather than an assert) i.e. this is not final or correct.

Reply:
Dask keys are NOT in general tuples - strings are common. However, for anything blockwise based (including all of dask-awkward), the pattern is (name: str, index: int). @douglasdavis can comment on whether this is also true for scalars.

In dask.core:
```
def iskey(key: object) -> bool:
    """Return True if the given object is a potential dask key; False otherwise."""
    typ = type(key)
    if typ is tuple:
        return all(iskey(i) for i in cast(tuple, key))
    return typ in {bytes, int, float, str}
```

Reply:
Hmm, OK! I wonder at what point can / should we encode this in our typing and function signatures?

Reply:
In this part of the code it does make sense for keys to always be a tuple in dask-awkward world, but when using dask-awkward and dask-histogram together, and since dask-histogram is dask-awkward-optimization-aware, it should be possible to run into a case where we have a string key (dask-histogram collections use a simple string as the collection key). Do you think this check is necessary?

Reply:
@douglasdavis if this is @ me, then I don't think _this_ check is right, but we probably need some kind of switch to determine whether `k[0]` will throw (i.e. if the key is not a sequence). I also imagine that `k[0]` is not intended to pull the first character of the key if it's a string. So, perhaps better
```suggestion
    required_layers = {k[0] if (isinstance(k, tuple) and ...) else k for k in keys if isinstance(k, tuple)}
```
I'm spit-balling here, though.

Reply:
Ah! but this is here just for the purpose of awkward blockwise layers, so we do always expect to see `(name, i)`. So the isinstance tuple check here makes sense and perhaps we should rename the variable to be `required_blockwise_layers`